### ISSUE TITLE
Expand Learning rate scheduling to any optimization hyperparameter

### ISSUE THREAD

**Issue by PhilippDahlinger** [pytorch/pytorch#83159]
### 🚀 The feature, motivation and pitch

I'm developing a new optimizer using the PyTorch optimizer framework. It does not depend on a learning rate, but on a KL-divergence. I would like to schedule other optimization hyperparameters (given in the optimizer config) with the PyTorch scheduler. Right now, the key `lr` is hardcoded inside the schedulers. 

I would like to have an option where you can specify the key/name of the scheduled variable when creating a scheduler. This would also unify other schedulers (for example weight decay). 

### Alternatives

I considered renaming my optimization hyperparameter `kl_div` to `lr` and use the learning rate scheduler. However, since it is not a learning rate this is not correct and may cause confusion on the user.
